### PR TITLE
fixed the token-permission and pinned-dependencies issue

### PR DESCRIPTION
**Pull request by harshitasao**
/area CI

**What this PR does / why we need it**:
fixed the token-permission and pinned-dependencies issue which are reported by [scorecard](https://scorecard.dev/viewer/?uri=github.com/falcosecurity/falco).

**Which issue(s) this PR fixes**:
Part of #3297 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

File: .github/workflows/reusable_test_packages.yaml [hunk 1]
```diff
@@ -54,7 +57,7 @@ jobs:
       - name: Run tests
         env:
           LSAN_OPTIONS: "intercept_tls_get_addr=0"
-        uses: falcosecurity/testing@main    
+        uses: falcosecurity/testing@32e319ae505fb330ae74db4502e605a5e517ff22 # main
```
Comment (FedeDP) [inline .github/workflows/reusable_test_packages.yaml:60]: Given this is our own repo we actually want to always point to latest main, not the current one. 
Can you revert this line?

Comment (FedeDP) [reply .github/workflows/reusable_test_packages.yaml:60]: /hold

File: .github/workflows/reusable_build_docker.yaml [hunk 2]
```diff
@@ -24,6 +24,9 @@ on:
 # then we upload all the tarballs to be later downloaded by reusable_publish_docker workflow.
 # In this way, we don't need to publish any arch specific image, 
 # and this "build" workflow is actually only building images.
+permissions:
```
Comment (sgaist) [inline .github/workflows/reusable_build_docker.yaml:27]: To keep space coherency

```suggestion

permissions:
```

File: .github/workflows/codespell.yml [hunk 3]
```diff
@@ -1,6 +1,9 @@
 name: Codespell
 on:
   pull_request:
+permissions:
```
Comment (sgaist) [inline .github/workflows/codespell.yml:4]: To keep space coherency

```suggestion

permissions:
```

**Review by sgaist — CHANGES_REQUESTED** [review]

**Review by FedeDP — APPROVED** [review]
/approve

**Comment by FedeDP** [comment]
cc @sgaist care to re-review? :D

**Comment by sgaist** [comment]
@FedeDP AFAICT, there's still the `uses: falcosecurity/testing@32e319ae505fb330ae74db4502e605a5e517ff22` line that you wanted reverted that is still present. Did you change your mind about it ?

**Comment by FedeDP** [comment]
@sgaist i double checked and it seems reverted to me! Am I missing anything? 😆

**Review by sgaist — APPROVED** [review]
My bad, I mixed that with the actions-github-release version.

**Comment by FedeDP** [comment]
This PR broke master CI triggering :/ don't know why.

**Pull request merged**